kmcd-prelude.md

```diff
@@ -233,11 +234,20 @@ module KMCD-GEN
  // -----------------------------------
     rule <k> GenStep => GenStepLoad ~> GenStepReplace ... </k>
          <random> BS => tail(BS) </random>
+         <used-random> _ => Int2Bytes(head(BS), 1, BE) </used-random>
```

Comment:
Isn't `BS` already a list of bytes? Why `Int2Bytes` in `<used-random>`?

Reply:
We're appending the `head(BS) :: Int` to the end of `<used-random>` (because it's the next choice we're making). I can possibly use `substrBytes(BS, 0, 1)` instead, so that it remains in the `Bytes` type.

Reply:
I've factored out `headAsBytes` helper function, which now uses `substrBytes(BS, 0, 1)` and makes the sorts more obvious.